pkg/controller/pipeline/pipeline_controller.go

```diff
@@ -225,30 +224,19 @@ func (c *Controller) syncHandler(key string) error {
 					return err
 				}
 			}
-		} else if devopsClient.GetDevOpsStatusCode(err) != http.StatusNotFound {
-			klog.Error(err, fmt.Sprintf("failed to get copyPipeline %s ", key))
-			return err
 		} else {
 			_, err := c.devopsClient.CreateProjectPipeline(nsName, copyPipeline)
 			if err != nil {
-				klog.Error(err, fmt.Sprintf("failed to get copyPipeline %s ", key))
+				klog.Error(err, fmt.Sprintf("failed to create copyPipeline %s ", key))
 				return err
 			}
 		}
 
 	} else {
 		// Finalizers processing logic
 		if sliceutil.HasString(copyPipeline.ObjectMeta.Finalizers, devopsv1alpha3.PipelineFinalizerName) {
-			_, err := c.devopsClient.GetProjectPipelineConfig(nsName, pipeline.Name)
-			if err != nil && devopsClient.GetDevOpsStatusCode(err) != http.StatusNotFound {
-				klog.Error(err, fmt.Sprintf("failed to get pipeline %s ", key))
-				return err
-			} else if err != nil && devopsClient.GetDevOpsStatusCode(err) == http.StatusNotFound {
-			} else {
-				if _, err := c.devopsClient.DeleteProjectPipeline(nsName, pipeline.Name); err != nil {
-					klog.Error(err, fmt.Sprintf("failed to delete pipeline %s in devops", key))
-					return err
-				}
+			if _, err := c.devopsClient.DeleteProjectPipeline(nsName, pipeline.Name); err != nil {
+				klog.Error(err, fmt.Sprintf("failed to delete pipeline %s in devops", key))
```

Comment:
Should we just ignore the error?  If we just ignore the error, I think the pipeline probably will be stuck there forever.